drivers/sensors/sensor.c

```diff
@@ -867,6 +720,6 @@ void sensor_unregister(FAR struct sensor_lowerhalf_s *lower, int devno)
   nxsem_destroy(&upper->exclsem);
   nxsem_destroy(&upper->buffersem);
 
-  sensor_buffer_release(upper->buffer);
+  circ_buf_uninit(&upper->buffer);
```

Comment:
Same comment as in previous PR: this memory should be allocated/allocated on first open()/ last close().

Reply:
okay, i will optimize code. Thank you .